wasm/src/programs/manager/execute.rs

```diff
@@ -57,10 +56,9 @@ impl ProgramManager {
     #[wasm_bindgen(js_name = executeFunctionOffline)]
     #[allow(clippy::too_many_arguments)]
     pub async fn execute_function_offline(
-        &mut self,
-        private_key: PrivateKey,
-        program: String,
-        function: String,
+        private_key: &PrivateKey,
+        program: &str,
+        function: &str,
         inputs: Array,
         prove_execution: bool,
         cache: bool,
```

Comment:
And this should be removed too.

Reply:
The cache is kept around in this function & this function only - reason described above.